manifests/fact.pp

```diff
@@ -4,27 +4,34 @@
 #
 define facter::fact (
   $value,
-  $fact      = $name,
-  $file      = 'facts.txt',
```

Comment:
Would prefer if the default values stay here and then the hiera data for windows can override. That way you don't have to use `Optional` to set `undef` since they are not really optional. It is also a pattern I use elsewhere.